firebase-functions/src/main/java/com/google/firebase/functions/HttpsCallableReference.kt

```diff
@@ -0,0 +1,159 @@
+   * types.
+   *  * [Map&amp;lt;String, ?&amp;gt;&gt;][java.util.Map], where the values are also one of these
+   * types.
+   *  * [org.json.JSONArray]
+   *  * [org.json.JSONObject]
+   *  * [org.json.JSONObject.NULL]
+   *
+   *
+   *
+   * If the returned task fails, the Exception will be one of the following types:
+   *
+   *
+   *  * [java.io.IOException] - if the HTTPS request failed to connect.
+   *  * [FirebaseFunctionsException] - if the request connected, but the function returned
+   * an error.
+   *
+   *
+   *
+   * The request to the Cloud Functions backend made by this method automatically includes a
+   * Firebase Instance ID token to identify the app instance. If a user is logged in with Firebase
+   * Auth, an auth token for the user will also be automatically included.
+   *
+   *
+   * Firebase Instance ID sends data to the Firebase backend periodically to collect information
+   * regarding the app instance. To stop this, see [ ][com.google.firebase.iid.FirebaseInstanceId.deleteInstanceId]. It will resume with a new
+   * Instance ID the next time you call this method.
+   *
+   * @param data Parameters to pass to the trigger.
+   * @return A Task that will be completed when the HTTPS request has completed.
+   * @see org.json.JSONArray
+   *
+   * @see org.json.JSONObject
+   *
+   * @see java.io.IOException
+   *
+   * @see FirebaseFunctionsException
+   */
+  fun call(data: Any?): Task<HttpsCallableResult?> {
+    return if (name != null) {
+      functionsClient.call(name, data, options)
+    } else {
+      functionsClient.call(url!!, data, options)
+    }
+  }
+
+  /**
+   * Executes this HTTPS endpoint asynchronously without arguments.
+   *
+   *
+   * The request to the Cloud Functions backend made by this method automatically includes a
+   * Firebase Instance ID token to identify the app instance. If a user is logged in with Firebase
+   * Auth, an auth token for the user will also be automatically included.
+   *
+   *
+   * Firebase Instance ID sends data to the Firebase backend periodically to collect information
+   * regarding the app instance. To stop this, see [ ][com.google.firebase.iid.FirebaseInstanceId.deleteInstanceId]. It will resume with a new
+   * Instance ID the next time you call this method.
+   *
+   * @return A Task that will be completed when the HTTPS request has completed.
+   */
+  fun call(): Task<HttpsCallableResult?> {
+    return if (name != null) {
+      functionsClient.call(name, null, options)
+    } else {
+      functionsClient.call(url!!, null, options)
+    }
+  }
+
+  /**
+   * Changes the timeout for calls from this instance of Functions. The default is 60 seconds.
+   *
+   * @param timeout The length of the timeout, in the given units.
+   * @param units The units for the specified timeout.
+   */
+  fun setTimeout(timeout: Long, units: TimeUnit) {
+    options.setTimeout(timeout, units)
+  }
+
+  val timeout: Long
```

Comment:
This created a field just for a getter. Is this desireable?

Reply:
yeah, in order to retain the same implicit calling in Kotlin- it needs to be a variable.

Reply:
This is going to cause us problems when we want to keep exposing values as `public static final` in Java, is there a workaround?

Reply:
I'm not sure I understand your question? Could you give me an example of where this would cause an issue?